src/utils.ts

```diff
@@ -150,3 +151,37 @@ export async function spawnAsync(executable: string, args: string[], cwd?: strin
     childProcess.on('exit', () => f(output));
   });
 }
+
+export async function resolveSourceMap(file: string, fileToSources: Map<string, string[]>, sourceToFile: Map<string, string>): Promise<string[]> {
+  if (!file.endsWith('.js'))
+    return [file];
+  const cached = fileToSources.get(file);
+  if (cached)
+    return cached;
+
+  const rl = readline.createInterface({ input: fs.createReadStream(file), crlfDelay: Infinity });
+
+  let lastLine: string | undefined;
+  rl.on('line', line => {
+    lastLine = line;
+  });
+  await new Promise(f => rl.on('close', f));
+
+  if (lastLine?.startsWith('//# sourceMappingURL=')) {
```

Comment:
what about inline sourcemaps? Not sure if they make sense here.

Reply:
We might need to support them at some point, but they are not the tsc/esbuild defaults.